### PR TITLE
[batch,query] Enable 8 core drivers with standard or highmem workers

### PR DESCRIPTION
**Pull request by jigold**
Stacked on #11397

### CODE REVIEW

**Comment by danking** [comment]
I pushed a test to this branch

**Comment by danking** [comment]
rebased, removed stacked

File: batch/batch/front_end/front_end.py [hunk 1]
```diff
@@ -770,10 +770,10 @@ async def _create_jobs(userdata: dict, job_specs: dict, batch_id: int, app: aioh
                     raise web.HTTPBadRequest(reason='cannot specify cpu and memory with machine_type')
 
                 if spec['process']['type'] == 'jvm':
-                    if 'cpu' in resources:
-                        raise web.HTTPBadRequest(reason='jvm jobs may not specify cpu')
-                    if 'memory' in resources and resources['memory'] != 'standard':
-                        raise web.HTTPBadRequest(reason='jvm jobs may not specify memory')
+                    if 'cpu' in resources and resources['cpu'] not in (1, 8):
+                        raise web.HTTPBadRequest(reason='invalid cpu for jvm jobs. must be 1 or 8')
+                    if 'memory' in resources and resources['memory'] == 'highcpu':
```
Comment (jigold) [inline batch/batch/front_end/front_end.py:775]: I messed up. This needs to be 'lowmem' I believe. Same with error msg below.

File: hail/python/hail/backend/service_backend.py [hunk 2]
```diff
@@ -334,7 +344,11 @@ async def _rpc(self,
                         iodir + '/out'
                     ],
                     mount_tokens=True,
-                    resources={'preemptible': False, 'memory': 'standard'}
+                    resources={
+                        'preemptible': False,
+                        'cpu': self.driver_cores or '1',
```
Comment (jigold) [inline hail/python/hail/backend/service_backend.py:349]: Can you double check what the type of 'cpu' and memory needs to be? I think we need to cast them as `str`.

File: batch/batch/front_end/front_end.py [hunk 3]
```diff
@@ -771,7 +771,7 @@ async def _create_jobs(userdata: dict, job_specs: dict, batch_id: int, app: aioh
 
                 if spec['process']['type'] == 'jvm':
                     if 'cpu' in resources and resources['cpu'] not in (1, 8):
-                        raise web.HTTPBadRequest(reason='invalid cpu for jvm jobs. must be 1 or 8')
+                        raise web.HTTPBadRequest(reason=f'invalid cpu for jvm jobs. must be 1 or 8 was {resources["cpu"]!r}')
```
Comment (jigold) [inline batch/batch/front_end/front_end.py:774]: I think the issue is the types. It should probably be `not in ('1', '8')`

**Review by danking — APPROVED** [review]

**Pull request merged**